src/garage/torch/policies/context_conditioned_policy.py

```diff
@@ -0,0 +1,268 @@
+# pylint: disable=attribute-defined-outside-init, protected-access
+"""A policy used in training meta reinforcement learning algorithms.
+
+It is used in PEARL (Probabilistic Embeddings for Actor-Critic Reinforcement
+Learning). The paper on PEARL can be found at https://arxiv.org/abs/1903.08254.
+Code is adapted from https://github.com/katerakelly/oyster.
+"""
+
+import numpy as np
+import torch
+from torch import nn
+import torch.nn.functional as F
+
+import garage.torch.utils as tu
+
+
+class ContextConditionedPolicy(nn.Module):
+    """A policy that outputs actions based on observation and latent context.
+
+    In PEARL, policies are conditioned on current state and a latent context
+    (adaptation data) variable Z. This inference network estimates the
+    posterior probability of z given past transitions. It uses context
+    information stored in the encoder to infer the probabilistic value of z and
+    samples from a policy conditioned on z.
+
+    Args:
+        latent_dim (int): Latent context variable dimension.
+        context_encoder (garage.torch.embeddings): Recurrent or
+            permutation-invariant context encoder.
+        policy (garage.torch.policies.Policy): Policy used to train the
+            network.
+        use_ib (bool): True if latent context is not deterministic; false
```

Comment:
please make this longer -- `use_information_bottleneck`. it's more typing, but makes the API more discoverable.